packages/features/schedules/components/Schedule.tsx

```diff
@@ -103,20 +103,24 @@ export const ScheduleDay = <TFieldValues extends FieldValues>({
       </div>
       <>
         {watchDayRange ? (
-          <div className="flex sm:ml-2">
-            <DayRanges
-              userTimeFormat={userTimeFormat}
-              labels={labels}
-              control={control}
-              name={name}
-              disabled={disabled}
-              className={{
-                dayRanges: className?.dayRanges,
-                timeRangeField: className?.timeRangeField,
-              }}
-            />
-            {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
-          </div>
+          <>
+            {watchDayRange.length > 0 && (
```

Comment:
How about we move this `watchDayRange.length > 0` check one level higher. So instead of adding another conditional inside of the dayRanges component maybe something like this 
`!!watchDayRange ? Day ranges component : Skeleton component`

Reply:
The `!!` will convert the `watchDayRange` variable to its boolean equivalent, similar to what `watchDayRange.length > 0` does. Or you can use `Boolean(watchDayRange)` but thats just a personal preference since both of them are the same

Reply:
Hey @Ryukemeister, yes, I agree with you, it really does seem like it needs to be written exactly like this.
```
{!!watchDayRange ? (
          <div className="flex sm:gap-2">
            <DayRanges
              userTimeFormat={userTimeFormat}
              labels={labels}
              control={control}
              name={name}
              disabled={disabled}
              className={{
                dayRanges: className?.dayRanges,
                timeRangeField: className?.timeRangeField,
              }}
            />
            {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
          </div>
        ) : (
          <SkeletonText className="ml-1 mt-2.5 h-6 w-48" />
        )}
```
     
but if you try to do that, you will see that in that case, we have one extra empty div

<img width="881" alt="image" src="https://github.com/calcom/cal.com/assets/3423341/75f368b3-dd9d-47b8-a3a0-1683816a60d1">

the same will be if I use `!!watchDayRange` instead of `watchDayRange.length > 0` 

The difference between these checks in this case:

1. watchDayRange.length > 0 
This line checks if the length of the watchDayRange array is greater than zero. This means there must be at least one element in the array for the condition to return true.
2. !!watchDayRange
This line first converts watchDayRange to a boolean value. If watchDayRange is an object or array (even if empty), !!watchDayRange will always return true because objects and arrays are "truthy" values in JavaScript. Thus, this condition checks for the existence of the array, not its emptiness.
The difference is that the first condition checks for the presence of elements within the array, while the second checks for the existence of the array itself, regardless of its content. If watchDayRange is an empty array ([]), the first condition will return false, while the second will return true.


And if I do:
```
     {watchDayRange.length > 0 ? (
          <div className="flex sm:gap-2">
            <DayRanges
              userTimeFormat={userTimeFormat}
              labels={labels}
              control={control}
              name={name}
              disabled={disabled}
              className={{
                dayRanges: className?.dayRanges,
                timeRangeField: className?.timeRangeField,
              }}
            />
            {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
          </div>
        ) : (
          <SkeletonText className="ml-1 mt-2.5 h-6 w-48" />
        )}
```
I will see a skeleton when I don't need to see it
<img width="312" alt="image" src="https://github.com/calcom/cal.com/assets/3423341/a58d222a-fc76-4ff8-be89-17fff57e019e">

So the best option here is to leave it as it is

Reply:
What if we do `watchDayRange && watchDayRange.length > 0`? That way day ranges only renders when watch day range is defined and its not empty. So what about  `watchDayRange && watchDayRange.length > 0 ? < DayRanges/> : < Skeleton/>`

Reply:
`watchDayRange && watchDayRange.length > 0 `
will give us `false` if we have `[] `and we will see a <Skeleton /> when we don't want to see it

so, code might look like that
```
    <>
        {!watchDayRange && <SkeletonText className="ml-1 mt-2.5 h-6 w-48" />}
        {watchDayRange.length > 0 && (
          <div className="flex sm:gap-2">
            <DayRanges
              userTimeFormat={userTimeFormat}
              labels={labels}
              control={control}
              name={name}
              disabled={disabled}
              className={{
                dayRanges: className?.dayRanges,
                timeRangeField: className?.timeRangeField,
              }}
            />
            {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
          </div>
        )}
      </>
      ```
      
      or like that
```

 <>
        {watchDayRange ? (
          <>
            {watchDayRange.length > 0 && (
              <div className="flex sm:gap-2">
                <DayRanges
                  userTimeFormat={userTimeFormat}
                  labels={labels}
                  control={control}
                  name={name}
                  disabled={disabled}
                  className={{
                    dayRanges: className?.dayRanges,
                    timeRangeField: className?.timeRangeField,
                  }}
                />
                {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
              </div>
            )}
          </>
        ) : (
          <SkeletonText className="ml-1 mt-2.5 h-6 w-48" />
        )}
      </>

```
  but not like that
```

        <>
        {watchDayRange && watchDayRange.length > 0 ? (
          <div className="flex sm:gap-2">
            <DayRanges
              userTimeFormat={userTimeFormat}
              labels={labels}
              control={control}
              name={name}
              disabled={disabled}
              className={{
                dayRanges: className?.dayRanges,
                timeRangeField: className?.timeRangeField,
              }}
            />
            {!!watchDayRange.length && !disabled && <div className="block">{CopyButton}</div>}
          </div>
        ) : (
          <SkeletonText className="ml-1 mt-2.5 h-6 w-48" />
        )}
      </>

```